src/ray/core_worker/pubsub/publisher.h

```diff
@@ -142,32 +169,54 @@ class Publisher {
   /// to it anymore.
   /// TODO(sang): Currently, clients don't send a RPC to unregister themselves.
   ///
-  /// \param subscriber_node_id The node id of the subscriber to unsubscribe.
+  /// \param subscriber_id The node id of the subscriber to unsubscribe.
   /// \return True if erased. False otherwise.
-  bool UnregisterSubscriber(const NodeID &subscriber_node_id);
+  bool UnregisterSubscriber(const SubscriberID &subscriber_id);
 
   /// Unregister subscription. It means the given object id won't be published to the
   /// subscriber anymore.
   ///
-  /// \param subscriber_node_id The node id of the subscriber.
+  /// \param subscriber_id The node id of the subscriber.
   /// \param object_id The object id of the subscriber.
   /// \return True if erased. False otherwise.
-  bool UnregisterSubscription(const NodeID &subscriber_node_id,
+  bool UnregisterSubscription(const SubscriberID &subscriber_id,
                               const ObjectID &object_id);
 
+  /// Check all subscribers, detect which subscribers are dead or its connection is timed
+  /// out, and clean up their metadata. This uses the goal-oriented logic to clean up all
+  /// metadata that can happen by subscriber failures. It is how it works;
+  /// - If there's no new long polling connection for the timeout, it refreshes the long
+  /// polling connection.
+  ///   If the subscriber is dead, there will be no new long polling connection coming in
+  ///   again.
```

Comment:
```suggestion
  ///   - If the subscriber is dead, there will be no new long polling connection coming in
  ///     again. Otherwise, the long polling connection will be reestablished.
```